<?xml version="1.0"?>
<launch>
    <!-- PID控制器节点 -->
    <node pkg="pid_control_pkg" type="position_pid_controller" name="position_pid_controller" output="screen" launch-prefix="bash -c 'sleep 5; $0 $@'">
        <!-- 控制参数 -->
        <param name="control_frequency" value="100.0" />
        
        <!-- 坐标系参数 -->
        <param name="map_frame" value="map" />
        <param name="laser_link_frame" value="laser_link" />

        <!-- 话题重映射 -->
        <remap from="/target_position" to="/target_position" />
        <remap from="/height" to="/height" />
        <remap from="/target_velocity" to="/target_velocity" />
    </node>
    <group ns="navigate">
        <param name="kp_xy" value="0.8" />
        <param name="ki_xy" value="0.0" />
        <param name="kd_xy" value="0.07" />
        <param name="kp_yaw" value="0.8" />
        <param name="ki_yaw" value="0.0" />
        <param name="kd_yaw" value="0.08" />
        <param name="kp_z" value="1.2" />
        <param name="ki_z" value="0.0" />
        <param name="kd_z" value="0.10" />
        <param name="max_linear_velocity" value="36.0" />//
        <param name="max_angular_velocity" value="30.0" />
        <param name="max_vertical_velocity" value="40.0" />
        <param name="min_linear_velocity" value="-36" />
        <param name="min_angular_velocity" value="-30" />
        <param name="min_vertical_velocity" value="-40" />
        <param name="Xpos_tolerance" value="4.0" />
        <param name="Ypos_tolerance" value="4.0" />
        <param name="yaw_tolerance" value="5.0" />
        <param name="height_tolerance" value="3.0" />
    </group>
    <group ns="fine_tune">
        <param name="kp_xy" value="0.3" />
        <param name="ki_xy" value="0.0" />
        <param name="kd_xy" value="0.1" />
        <param name="kp_yaw" value="1.0" />
        <param name="ki_yaw" value="0.0" />
        <param name="kd_yaw" value="0.2" />
        <param name="kp_z" value="0.5" />
        <param name="ki_z" value="0.0" />
        <param name="kd_z" value="0.15" />
        <param name="max_linear_velocity" value="6.0" />
        <param name="max_angular_velocity" value="-6.0" />
        <param name="max_vertical_velocity" value="6.0" />
        <param name="min_linear_velocity" value="-6.0" />
        <param name="min_angular_velocity" value="6.0" />
        <param name="min_vertical_velocity" value="-6.0" />
        <param name="Xpos_tolerance" value="6" />
        <param name="Ypos_tolerance" value="20" />
        <param name="yaw_tolerance" value="5.0" />
        <param name="height_tolerance" value="6" />
    </group>
</launch>